.github/workflows/mvn_docker_image.yml

```diff
@@ -52,11 +52,20 @@ on:
                 default: local
                 type: string
                 description: Currently only used for native builds
+            container_registry_type:
+                required: false
+                default: 'azure-cr'
+                type: string
+                description: "the type of container registry. Supports 'azure-cr', 'amazon-ecr'"
             runs_on:
                 required: false
                 default: ubuntu-latest
                 type: string
                 description: Set to 'self-hosted' if needed.
+            ecr_region:
+                required: false
+                type: string
+                description: required if input container_registry_type is 'amazon-ecr'
 
         secrets:
             artifactory_user:
```

Comment:
Maybe also move non-secrets to input section? E.g. `artifactory_url`, `artifactory_repo`, `acr_registry`,...

Reply:
This would be a **very** breaking change

Reply:
`v3`? :D

Reply:
It's an unnecessary change with no real benefits since will not set these values as defaults in a public repo

Reply:
Fair point